contracts/VestedEscrow.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+/*
+Rewrite of Curve Finance's Vested Escrow
+found at https://github.com/curvefi/curve-dao-contracts/blob/master/contracts/VestingEscrow.vy
+
+Changes:
+- no disable methods
+- only one fund admin
+- add claim and stake
+*/
+import "./helper/MathUtil.sol";
+import "./Interfaces/IRewards.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+
+contract VestedEscrow is ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IERC20 public rewardToken;
+    address public admin;
+    address public fundAdmin;
+    address public stakeContract;
+
+    uint256 public startTime;
+    uint256 public endTime;
+    uint256 public totalTime;
+    uint256 public initialLockedSupply;
+    uint256 public unallocatedSupply;
+
+    mapping(address => uint256) public initialLocked;
+    mapping(address => uint256) public totalClaimed;
+
+    address[] public extraRewards;
+
+    event Fund(address indexed recipient, uint256 reward);
+    event Claim(address indexed user, uint256 amount);
+
+    constructor(
+        address rewardToken_,
+        uint256 starttime_,
+        uint256 endtime_,
+        address stakeContract_,
+        address fundAdmin_
+    ) {
+        require(starttime_ >= block.timestamp, "start must be future");
+        require(endtime_ > starttime_, "end must be greater");
+
+        rewardToken = IERC20(rewardToken_);
+        startTime = starttime_;
+        endTime = endtime_;
+        totalTime = endTime.sub(startTime);
```

Comment:
@c-future when I try to use trySub I get this error `TypeError: Type tuple(bool,uint256) is not implicitly convertible to expected type uint256.`. Are you sure I should use trySub? I don't see it being used anywhere else in our contracts

Reply:
yes , it will be a future task so keep it for now , I will customize the openzipplein safemath lib to make it easy to use, then we can update the math then